vitessce/vitessce
# Unit test coverage

Fixes #145
Fixes #1599
Fixes #1498
Fixes #1559
Fixes #1512


#### Background


#### Change List
- Upgrade vitest
  - Use vitest-canvas-mock
    - [Pin jest-canvas-mock version](https://github.com/wobsoriano/vitest-canvas-mock/issues/5#issuecomment-1559275559)
  - Explicitly import `describe, it, expect` in unit tests
- Use vitest workspaces; run unit tests once at root of repo rather than independently per sub-package https://vitest.dev/guide/workspace.html
- Configure coverage reporting in `vite.config.js`
- Add github action to comment on PRs with any changes in coverage
- Upgrade pnpm setup github action (Note: not directly related to the addition of unit test coverage)
- Specify `shell: bash` to fix unit test execution on Windows in github actions
- Clean up `test.sh` script
- Clean up `packages/vit-s/schemas` JSON schema files that are no longer used
- Upgrade to PNPM v8 in README and github actions
  - [Revert pnpm-lock lockfile version](https://github.com/pnpm/pnpm/issues/6648#issuecomment-1588863457)
#### Checklist
 - [ ] Ensure PR works with all demos on the dev.vitessce.io homepage
 - [ ] Open (draft) PR's into [`vitessce-python`](https://github.com/vitessce/vitessce-python) and [`vitessce-r`](https://github.com/vitessce/vitessce-r) if this is a release PR
